build_system/src/test.rs

```diff
@@ -1003,54 +1004,153 @@ where
 }
 
 fn test_rustc(env: &Env, args: &TestArg) -> Result<(), String> {
-    test_rustc_inner(env, args, |_| Ok(false))
+    test_rustc_inner(env, args, |_| Ok(false), "run-make")?;
+    test_rustc_inner(env, args, |_| Ok(false), "ui")
 }
 
 fn test_failing_rustc(env: &Env, args: &TestArg) -> Result<(), String> {
-    test_rustc_inner(env, args, |rust_path| {
-        // Removing all tests.
-        run_command(
-            &[
-                &"find",
-                &"tests/ui",
-                &"-type",
-                &"f",
-                &"-name",
-                &"*.rs",
-                &"-not",
-                &"-path",
-                &"*/auxiliary/*",
-                &"-delete",
-            ],
-            Some(rust_path),
-        )?;
-        // Putting back only the failing ones.
-        let path = "tests/failing-ui-tests.txt";
-        if let Ok(files) = std::fs::read_to_string(path) {
-            for file in files.split('\n').map(|line| line.trim()).filter(|line| !line.is_empty()) {
-                run_command(&[&"git", &"checkout", &"--", &file], Some(&rust_path))?;
+    test_rustc_inner(
+        env,
+        args,
+        prepare_files_callback_failing("tests/failing-run-make-tests.txt", "run-make"),
+        "run-make",
+    )?;
+    test_rustc_inner(
+        env,
+        args,
+        prepare_files_callback_failing("tests/failing-ui-tests.txt", "ui"),
+        "ui",
+    )
```

Comment:
It seems the failing [UI tests are not ran in the CI](https://github.com/antoyo/rustc_codegen_gcc/actions/runs/8409674523/job/23027225505?pr=7#step:15:2919).

Please check if this is the case and fix the issue if there's one.

Reply:
@antoyo Ok so i got around 170 tests on them, should I remove the pattern and add those to failing test file?

Reply:
What I mean is that this should run the failing `make-run` tests in addition to the failing `ui` tests, but it seems the `ui` tests are not ran in the CI.

Perhaps the function does an early return because of `?` so the `ui` tests are not ran.